index.js

```diff
@@ -379,24 +380,39 @@ function Argv (processArgs, cwd) {
 
   var versionOpt = null
   self.version = function (opt, msg, ver) {
-    if (arguments.length === 1) {
+    if (arguments.length === 0) {
+      ver = guessVersion()
+      opt = 'version'
+    } else if (arguments.length === 1) {
       ver = opt
-      versionOpt = 'version'
-      msg = usage.deferY18nLookup('Show version number')
+      opt = 'version'
     } else if (arguments.length === 2) {
-      versionOpt = opt
       ver = msg
-      msg = usage.deferY18nLookup('Show version number')
-    } else {
-      versionOpt = opt
     }
 
+    versionOpt = opt
+    msg = msg || usage.deferY18nLookup('Show version number')
+
     usage.version(ver || undefined)
     self.boolean(versionOpt)
     self.describe(versionOpt, msg)
     return self
   }
 
+  function guessVersion () {
+    var obj = readPkgUp.sync({
+      cwd: path.resolve(__dirname, '../')
+    })
```

Comment:
I think the only time this may return an incorrect result is if yargs is a nested dependency - but that shouldn't really happen, right?